compute_sdk/globus_compute_sdk/serialize/facade.py

```diff
@@ -15,35 +23,122 @@
+    t.Union[type[SerializationStrategy], str, AllowlistWildcard]
+]
+
+
+def assert_strategy_type_valid(
+    strategy_type: type[SerializationStrategy], for_code: bool
+) -> None:
+    if strategy_type not in SELECTABLE_STRATEGIES:
+        raise SerdeError(
+            f"{strategy_type.__name__} is not a known serialization strategy"
+            f" (must be one of {SELECTABLE_STRATEGIES})"
+        )
+
+    if strategy_type.for_code != for_code:
+        etype = "code" if for_code else "data"
+        gtype = "code" if strategy_type.for_code else "data"
+        raise SerdeError(
+            f"{strategy_type.__name__} is a {gtype} serialization strategy,"
+            f" expected a {etype} strategy"
+        )
+
+
+def validate_strategy(
+    strategy: SerializationStrategy, for_code: bool
+) -> SerializationStrategy:
+    assert_strategy_type_valid(type(strategy), for_code)
+    return strategy
+
+
+def parse_allowlist(
+    unvalidated: DeserializerAllowlist | None,
+) -> set[type[SerializationStrategy]]:
+    if not unvalidated:
+        return set()
+
+    validated: set[type[SerializationStrategy]] = set()
+    wildcards: set[AllowlistWildcard] = set()
+    for value in unvalidated:
+        try:
+            AllowlistWildcard(value)
+        except ValueError:
+            pass
+        else:
+            # value is either an instance of AllowlistWildcard or the .value of
+            # such an instance. mypy is not smart enough to know that
+            wildcards.add(value)  # type: ignore
+            continue
+
+        resolved_strategy_class = None
+        if isinstance(value, str):
+            try:
+                mod_name, class_name = value.rsplit(".", 1)
+                mod = importlib.import_module(mod_name)
+                resolved_strategy_class = getattr(mod, class_name)
+            except Exception as e:
+                raise SerdeError(f"`{value}` is not a valid path to a strategy") from e
+        else:
+            resolved_strategy_class = value
+
+        if not inspect.isclass(resolved_strategy_class) or not issubclass(
+            resolved_strategy_class, SerializationStrategy
+        ):
+            raise SerdeError(
+                "Allowed deserializers must either be SerializationStrategies"
+                f" or valid paths to them (got {value})"
+            )
+
+        assert_strategy_type_valid(
+            resolved_strategy_class, resolved_strategy_class.for_code
+        )
+        validated.add(resolved_strategy_class)
+
+    for wildcard in wildcards:
+        for_code = wildcard == AllowlistWildcard.CODE
+        if any(s.for_code == for_code for s in validated):
+            raise SerdeError(
+                f"Cannot mix '{wildcard}' with specific deserializers"
+                f" (got {list(unvalidated)})"
+            )
+        validated.update(s for s in SELECTABLE_STRATEGIES if s.for_code == for_code)
+
+    if len({s.for_code for s in validated}) != 2:
+        raise SerdeError(
```

Comment:
Fun!  My brain would have done this differently.  The (longer) approach in my head until I read this was:
```python
if {True, False} == {s.for_code for s in validated}:
```
(No action needed, just observing brain differences.)